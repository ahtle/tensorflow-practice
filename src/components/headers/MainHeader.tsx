import { Link } from 'react-router-dom';

const MainHeader = () => {
    return (
        <div id="MainHeader" className="bg-black text-white">
            <div className="container mx-auto p-2">
                <div id="logo-container">
                    <Link className="flex items-center" to='/'>
                        <img src={`${process.env.PUBLIC_URL}/images/logo192.png`} alt="React logo" width="25"/>
                        <p className="text-xl mx-2"><span className="text-react-color">React</span> x <span className="text-tensorflow-color">TensorFlow</span></p>
                        <img src={`${process.env.PUBLIC_URL}/images/tf-logo.png`} alt="TensorFlow logo" width="25"/>
                    </Link>
                </div>
            </div>
        </div>
    )
};

export default MainHeader;