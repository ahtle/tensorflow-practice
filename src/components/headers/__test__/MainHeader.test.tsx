import { shallow } from 'enzyme';
import MainHeader from '../MainHeader';

describe("<MainHeader />", () => {
    it("renders MainHeader without crashing", () => {
        const component = shallow(<MainHeader />);
        expect(component).toMatchSnapshot();
    });
    
    it("render images", () => {
        const component = shallow(<MainHeader />);
        const reactPath = `${process.env.PUBLIC_URL}/images/logo192.png`;
        const tfPath = `${process.env.PUBLIC_URL}/images/tf-logo.png`;
        expect(component.find('img[data-testId="react-logo"]').prop("src")).toEqual(reactPath);
        expect(component.find('img[data-testId="tf-logo"]').prop("src")).toEqual(tfPath);
    });
});