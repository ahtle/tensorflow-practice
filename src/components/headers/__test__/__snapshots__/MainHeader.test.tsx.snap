// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainHeader /> renders MainHeader without crashing 1`] = `ShallowWrapper {}`;
